### PR TITLE
v0.37.0, support for source-based xcframeworks

### LINKED ISSUE #3097: Unrecognized arguments: --use-xcframeworks (carthage update --use-xcframeworks)
* carthage install method: [ ] .pkg, [x] homebrew, [ ] source
* `which carthage`: /usr/local/bin/carthage
* `carthage version`: 0.36.0
* `xcodebuild -version`: Xcode 12.3; Build version 12C33
* Are you using `--no-build`? no
* Are you using `--no-use-binaries`?  no
* Are you using `--use-submodules`? no
* Are you using `--cache-builds`? no
* Are you using `--new-resolver`? no
* `macOS version`: Catalina 10.15.7

**Cartfile**
```
github "ReactiveX/RxSwift"
github "RxSwiftCommunity/RxDataSources"
github "SnapKit/SnapKit"
github "scenee/FloatingPanel"
github "pkluz/PKHUD"
github "layoutBox/PinLayout"

binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseAnalyticsBinary.json"
binary "https://dl.google.com/dl/firebase/ios/carthage/FirebaseCrashlyticsBinary.json"

github "yagiz/Bagel"
```

**Carthage Output**
```
$ carthage update --use-xcframeworks
Unrecognized arguments: --use-xcframeworks
```

**Actual outcome**
`Unrecognized arguments: --use-xcframeworks`

**Expected outcome**
Frameworks have been successfully pulled into the project

### LINKED ISSUE #3102: Release a new version of Carthage with the XCFrameworks change!
Master was updated 17 days ago without a new release actually being drafted and released. Could someone please see to it that the release happens, or, perhaps give privileges over the repo to me, so I can do it? Thanks!

### PR DESCRIPTION
**Pull request by elliottwilliams**
Here's a branch to base the next release off of. I'm not aware of any other code changes we need to make, but @tmspzz let me know if I've missed something :)

Closes #3102, closes #3097

### CODE REVIEW

**Comment by elliottwilliams** [comment]
Added a one-line fix to allow xcframework builds to work when a user's Carthage/Build directory isn't on the same volume where the xcframework was built. Sanity checked that these lines still work as expected.

**Comment by tmspzz** [comment]
LGTM

**Review by tmspzz — APPROVED** [review]

**Review by Vitalii-Gozhenko — APPROVED** [review]

**Review by Cubillos003 — APPROVED** [review]

**Comment by winstondu** [comment]
@elliottwilliams , please release asap.

**Comment by ferologics** [comment]
@elliottwilliams how I can test these changes on my M1 mac?

**Comment by elliottwilliams** [comment]
> @elliottwilliams how I can test these changes on my M1 mac? Should I just refer to the [Building platform-independent xcframeworks](https://github.com/Carthage/Carthage#building-platform-independent-xcframeworks-xcode-12-and-above) section?

@ferologics you'll need to install this unreleased version of carthage first, but then yes, that section is how you'll build using xcframeworks.

To test this exact PR, you can check out [its branch](https://github.com/elliottwilliams/Carthage/tree/release_0.37.0) and run `make install`.

If you just want to test xcframework support, building from master is fine. You can do that using `brew install --HEAD carthage`.

**Comment by ferologics** [comment]
@elliottwilliams tested and all my dependencies look like they work fine. (With the exception of [Mapbox](https://github.com/mapbox/mapbox-gl-native-ios/issues/171))

Thanks 🙏

**Comment by nuno-vieira** [comment]
If XCFrameworks is the future, and it is required to work with Xcode12+. Isn't it a better idea for it to be the default? 🤔

**Comment by elliottwilliams** [comment]
@ferologics tysm for the testing! I'm wrapping up drafting release notes, and will plan to release Monday. I don't wanna be on the hook this weekend if we break something dramatically and need to respond.

> If XCFrameworks is the future, and it is required to work with Xcode12+. Isn't it a better idea for it to be the default? 🤔

@nuno-vieira discussed: https://github.com/Carthage/Carthage/pull/3071#discussion_r529600656 
tl;dr, compatibility with existing workflows is top priority, _including_ existing workflows that use the workaround xcconfig. This version will give you a meaningful error message if building fails because of overlapping architectures. Definitely open to revisiting in the future.

File: README.md [hunk 1]
```diff
@@ -75,15 +76,34 @@ Once you have Carthage [installed](#installing-carthage), you can begin adding f
...
 
 1. Create a [Cartfile][] that lists the frameworks you’d like to use in your project.
-1. Run `carthage update --use-xcframeworks`. This will fetch dependencies into a [Carthage/Checkouts][] folder and build each one or download a pre-compiled xcframework.
-1. On your application targets’ _General_ settings tab, in the _Embedded Binaries_ section, drag and drop each xcframework you want to use from the [Carthage/Build][] folder on disk.
+1. Run `carthage update --use-xcframeworks`. This will fetch dependencies into a [Carthage/Checkouts][] folder and build each one or download a pre-compiled XCFramework.
+1. On your application targets’ _General_ settings tab, in the _Frameworks, Libraries, and Embedded Content_ section, drag and drop each XCFramework you want to use from the [Carthage/Build][] folder on disk.
+
+##### Migrating a project from framework bundles to XCFrameworks
+
+We encourage using XCFrameworks as of version 0.37.0 (January 2021), and require XCFrameworks when building on an Apple Silicon Mac. Switching from discrete framework bundles to XCFrameworks requires a few changes to your project:
```
Comment (Tovkal) [inline README.md:87]: Should it say February 2021 instead of January?

**Comment by vincentisambart** [comment]
By the way xcframeworks themselves still seem to have problems.
From what I tested it seems Xcode's build system doesn't handle properly xcframeworks only used in test targets (you can work around that adding these xcframeworks to your app or to a framework target that your test target depends on).

**Review by jdhealy — APPROVED** [review]

**Comment by jadar** [comment]
> Mapbox

@ferologics Have you found a way to include Mapbox with the xcframeworks? It's the only thing keeping me from adopting them.

**Comment by ferologics** [comment]
@jadar no 🙃 The issue seems to be neglected on their side. It looks like somebody has to step up and do the needful. I will poke around MapBox [development guide](https://github.com/mapbox/mapbox-maps-ios/blob/main/DEVELOPING.md) this weekend and maybe get something working.

**Comment by Devkey19** [comment]
[]()https://github.com/Carthage/Carthage/commit/0668de43eb5d323d2e816eaab83677f50a8eeb24 bug[]()

File: README.md [hunk 2]
```diff
@@ -10,7 +10,8 @@ Carthage builds your dependencies and provides you with binary frameworks, but y
 - [Installing Carthage](#installing-carthage)
 - [Adding frameworks to an application](#adding-frameworks-to-an-application)
 	- [Getting started](#getting-started)
-		- [Building platform-independent xcframeworks](#building-platform-independent-xcframeworks-xcode-12-and-above)
```
Comment (Devkey19) [inline README.md:13]: []()

**Review by Devkey19 — APPROVED** [review]
[]()
